src/main/scala/tech/cryptonomic/conseil/tezos/repositories/TezosRepositories.scala

```diff
@@ -0,0 +1,190 @@
+package tech.cryptonomic.conseil.tezos.repositories
+
+import tech.cryptonomic.conseil.tezos.Fees
+import tech.cryptonomic.conseil.tezos.Fees.AverageFees
+import tech.cryptonomic.conseil.tezos.TezosTypes.{
+  Account,
+  AccountId,
+  Block,
+  BlockHash,
+  BlockReference,
+  BlockTagged,
+  ContractId,
+  Delegate,
+  PublicKeyHash,
+  Voting
+}
+
+/** Defines repository operations related to chain blocks
+  * Parametric on the returned effect type wrapper
+  */
+trait BlocksRepository[Eff[_]] {
+
+  /** is there any block stored at all? */
+  def anyBlockAvailable: Eff[Boolean]
+
+  /** Checks if a block for this hash is stored on db
+    * @param hash Identifies the block
+    * @return     true if block exists
+    */
+  def blockExists(hash: BlockHash): Eff[Boolean]
+
+  /** Computes the max level of blocks.
+    * @return the max level found or none
+    */
+  def fetchMaxBlockLevel: Eff[Option[Int]]
+
+  /** Fetch information on a block related to it being part
+    * of a fork in the chain.
+    * Invalid meaning detected on a fork which has been rejected currently.
+    * Semantic is
+    * - None: block was never marked invalid
+    * - Some(true): block is currently invalid
+    * - Some(false): block was invalid but is not anymore
+    *
+    * This should be restored when the forking is handled by Lorre
+    */
+  //def blockIsInvalidated(hash: BlockHash): Eff[Option[Boolean]]
```

Comment:
Commented code. 

If you're going to resurrect that method I would suggest changing its name to `blockIsValid` in order to avoid the double negative.

Reply:
It has a rather precise semantic, for now it's more correct because it tells you if it's been invalidated at least once in the past (not necessarily if it is still invalid)